rest-service/manager_rest/resource_manager.py

```diff
@@ -580,6 +583,37 @@ def delete_deployment(self,
         # Delete deployment data  DB (should only happen AFTER the workflow
         # finished successfully, hence the delete_db_mode flag)
         else:
+            # check for external targets
+            deployment_dependencies = self.sm.list(
+                models.InterDeploymentDependencies,
+                filters={'source_deployment': deployment})
+            external_targets = set(
+                json.dumps(x.external_target) for x in
+                deployment_dependencies if x.external_target)
+            if not external_targets:
+                return self.sm.delete(deployment)
+
+            # if external targets found - delete them from IDD in ext. managers
+            manager_ips = [x.private_ip for x in self.sm.list(models.Manager)]
+            external_source = {
+                'deployment': deployment.id,
+                'tenant': deployment.tenant_name,
+                'host': manager_ips
+            }
+            for target in external_targets:
+                target_client_config = json.loads(target)['client_config']
+                external_client = CloudifyClient(**target_client_config)
+                dep = external_client.inter_deployment_dependencies.list()
+
+                dep = [d for d in dep if
```

Comment:
Maybe use a different variable name here, since we already have `dep`?

Reply:
1